src/stream/src/executor/managed_state/top_n/top_n_state.rs

```diff
@@ -31,27 +33,46 @@ use crate::executor::top_n::TopNCache;
 pub struct ManagedTopNState<S: StateStore> {
     /// Relational table.
     pub(crate) state_table: StateTable<S>,
-    /// For deserializing `OrderedRow`.
-    ordered_row_deserializer: OrderedRowSerde,
+
+    /// Used for serializing pk into CacheKey.
+    first_key_serde: OrderedRowSerde,
+    second_key_serde: OrderedRowSerde,
```

Comment:
```suggestion
    cache_key_serde: (OrderedRowSerde, OrderedRowSerde),
```
might be better.

Similar for other places